lib/react.js

```diff
@@ -58,7 +58,7 @@ declare class ReactComponent<D, P, S> {
  * existential type (*). The * can be thought of as an "auto" instruction to the
  * typechecker, telling it to fill in the type from context.
  */
-type ReactClass<D, P, S> = _ReactClass<D, P, S, *>;
+type ReactClass<D, P, S> = _ReactClass<$ObjTest<{},D>, P, S, *>;
```

Comment:
Clever idea. But I'm not convinced this is the best way to solve this problem. See below.